File: Obj-C-Astronomy/Obj-C-Astronomy/Model/Rover.m
//
//  Rover.m
//  Obj-C-Astronomy
//
//  Created by Patrick Millet on 6/16/20.
//  Copyright © 2020 PatrickMillet79. All rights reserved.
//

#import "Rover.h"

@implementation Rover
- (instancetype)initWithIdentifier:(int)identifier
                              name:(NSString *)name
                       landingDate:(NSString *)landingDate
                        launchDate:(NSString *)launchDate
                            status:(NSString *)status
                            maxSol:(int)maxSol
                           maxDate:(NSString *)maxDate
                       totalPhotos:(int)totalPhotos
                            sols:(NSArray *)sols
{
    if (self = [super init]) {
         _identifier = identifier;
         _name = name;
         _landingDate = landingDate;
         _launchDate = launchDate;
         _status = status;
         _maxSol = maxSol;
         _maxDate = maxDate;
         _totalPhotos = totalPhotos;
         _sols = sols;
     }
     
     return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dictionary
{
    NSString *name = [dictionary objectForKey:@"name"];
    NSString *landingDate = [dictionary objectForKey:@"landing_date"];
    NSString *launchDate = [dictionary objectForKey:@"launch_date"];
    NSString *status = [dictionary objectForKey:@"status"];
    NSNumber* maxSol = [dictionary objectForKey:@"max_sol"];
    NSString *maxDate = [dictionary objectForKey:@"max_date"];
    NSNumber* totalPhotos = [dictionary objectForKey:@"total_photos"];

    NSArray *photosArray = [dictionary objectForKey:@"photos"];

    NSMutableArray *sols = [[NSMutableArray alloc] init];

    for (NSDictionary *photoDictionary in photosArray) {
        NSNumber *sol = [photoDictionary valueForKey:@"sol"];
        [sols addObject:sol];
    }

    return [self initWithIdentifier:0
                               name:name
                        landingDate:landingDate
                         launchDate:launchDate
                             status:status
                             maxSol:maxSol.intValue
                            maxDate:maxDate
                        totalPhotos:totalPhotos.intValue
                               sols: sols];
}

@end

File: Astronomy Objc/Astronomy Objc/Models/Rover.swift
//
//  Rover.swift
//  Astronomy Objc
//
//  Created by Vincent Hoang on 7/20/20.
//  Copyright © 2020 Vincent Hoang. All rights reserved.
//

import UIKit

@objc class Rover: NSObject {
    @objc let id: Int
    @objc let name: String
    @objc let landing_date: String
    @objc let launch_date: String
    @objc let status: String
    
    @objc init(_ id: Int, _ name: String, _ landing_date: String, _ launch_date: String, _ status: String) {
        self.id = id
        self.name = name
        self.landing_date = landing_date
        self.launch_date = launch_date
        self.status = status
    }
}
